hoodie-client/src/main/java/com/uber/hoodie/table/HoodieCopyOnWriteTable.java

```diff
@@ -626,9 +627,14 @@ private PartitionCleanStat merge(PartitionCleanStat other) {
     private Boolean deleteFileAndGetResult(String deletePathStr) throws IOException {
         Path deletePath = new Path(deletePathStr);
         logger.debug("Working on delete path :" + deletePath);
-        boolean deleteResult = getFs().delete(deletePath, false);
-        if (deleteResult) {
-            logger.debug("Cleaned file at path :" + deletePath);
+        //handle wildcard matches, especially for log* files with versions
+        FileStatus [] files = getFs().globStatus(deletePath);
```

Comment:
Worried about the extra globStatus.. the FileSlice also has the log files. lets just reimplement without the extra call or special casing.